clang/lib/CodeGen/CGHLSLRuntime.cpp

```diff
@@ -379,6 +380,18 @@ llvm::Value *CGHLSLRuntime::emitInputSemantic(IRBuilder<> &B,
                                               const ParmVarDecl &D,
                                               llvm::Type *Ty) {
   assert(D.hasAttrs() && "Entry parameter missing annotation attribute!");
+
+  if (D.hasAttr<HLSLSV_GroupIDAttr>()) {
+    if (getArch() == llvm::Triple::spirv) {
+      llvm::Type *Ty = CGM.getTypes().ConvertTypeForMem(D.getType());
+      llvm::Function *IntrinsicID =
+          CGM.getIntrinsic(Intrinsic::spv_load_builtin, {Ty});
+      return B.CreateCall(IntrinsicID, {B.getInt32(/* WorkgroupID */ 26)});
+    } else
+      // FIXME: getIntrinsic(getGroupIDIntrinsic())
+      return nullptr;
+  }
```

Comment:
Could something be added to `CommonSPIRTargetCodeGenInfo` and the DXIL equivalent? That way the spir-v specific ids will be hidden in a spirv specific class.

Reply:
Yes, this can clearly be refactorized in a real implementation. Just wanted to keep it simple for the draft.